### PR TITLE
Feature/debug view part1

### PR DESCRIPTION
**Pull request by NicolaChimerapps**
Replaced ugly custom focus painter with platform defaults
Added extra icon to switch to debug view mode (not implemented)

### CODE REVIEW

File: niddler-plugin/src/main/java/com/icapps/niddler/ui/impl/IntelliJToolbar.kt [hunk 1]
```diff
@@ -62,60 +78,76 @@ class IntelliJToolbar(panel: SimpleToolWindowPanel) : NiddlerMainToolbar {
...
+    override fun isSelected(e: AnActionEvent?): Boolean {
+        return toolbar.mode == MODE_TIMELINE
+    }
 
-        if (toolbar.mode == 0) {
-            e.presentation.icon = selectedIcon
-        } else {
-            e.presentation.icon = defaultIcon
+    override fun setSelected(e: AnActionEvent?, state: Boolean) {
+        if (state) {
```
Comment (vanlooverenkoen) [inline niddler-plugin/src/main/java/com/icapps/niddler/ui/impl/IntelliJToolbar.kt]: I would change the state variable to "isSelected"

File: niddler-plugin/src/main/java/com/icapps/niddler/ui/impl/IntelliJToolbar.kt [hunk 2]
```diff
@@ -62,60 +78,76 @@ class IntelliJToolbar(panel: SimpleToolWindowPanel) : NiddlerMainToolbar {
...
+                           private val defaultIcon: Icon) : ToggleAction("Linked mode",
+        "View messages in linked mode. Showing the request and response grouped together. When supported, this mode will also show the actual network requests", defaultIcon),
+        DumbAware {
+
+    override fun isSelected(e: AnActionEvent?): Boolean {
+        return toolbar.mode == MODE_LINKED
+    }
+
+    override fun setSelected(e: AnActionEvent?, state: Boolean) {
+        if (state) {
```
Comment (vanlooverenkoen) [inline niddler-plugin/src/main/java/com/icapps/niddler/ui/impl/IntelliJToolbar.kt]: I would change the state variable to "isSelected"

File: niddler-plugin/src/main/java/com/icapps/niddler/ui/impl/IntelliJToolbar.kt [hunk 3]
```diff
@@ -62,60 +78,76 @@ class IntelliJToolbar(panel: SimpleToolWindowPanel) : NiddlerMainToolbar {
...
-    override fun actionPerformed(e: AnActionEvent) {
-        toolbar.listener?.onLinkedSelected()
-        toolbar.mode = 1
-        toolbar.internal.updateActionsImmediately()
+    override fun isSelected(e: AnActionEvent?): Boolean {
+        return toolbar.mode == MODE_DEBUG
+    }
+
+    override fun setSelected(e: AnActionEvent?, state: Boolean) {
+        if (state) {
```
Comment (vanlooverenkoen) [inline niddler-plugin/src/main/java/com/icapps/niddler/ui/impl/IntelliJToolbar.kt]: I would change the state variable to "isSelected"

File: niddler-plugin/src/main/java/com/icapps/niddler/ui/impl/IntelliJToolbar.kt (hunk 1 above)
Comment (NicolaChimerapps) [reply niddler-plugin/src/main/java/com/icapps/niddler/ui/impl/IntelliJToolbar.kt]: This comes from the definition we are overriding. Best to keep the definition as close to the contract as possible

File: niddler-plugin/src/main/java/com/icapps/niddler/ui/impl/IntelliJToolbar.kt (hunk 2 above)
Comment (NicolaChimerapps) [reply niddler-plugin/src/main/java/com/icapps/niddler/ui/impl/IntelliJToolbar.kt]: This comes from the definition we are overriding. Best to keep the definition as close to the contract as possible

File: niddler-plugin/src/main/java/com/icapps/niddler/ui/impl/IntelliJToolbar.kt (hunk 3 above)
Comment (NicolaChimerapps) [reply niddler-plugin/src/main/java/com/icapps/niddler/ui/impl/IntelliJToolbar.kt]: This comes from the definition we are overriding. Best to keep the definition as close to the contract as possible

**Pull request merged**